src/service/action-impl.js

```diff
@@ -71,16 +75,19 @@ export class ActionInvocation {
    * @param {!Node} target
    * @param {string} method
    * @param {?JSONType} args
+   * @param {?string} argsExpression
    * @param {?Element} source
    * @param {?Event} event
    */
-  constructor(target, method, args, source, event) {
+  constructor(target, method, args, argsExpression, source, event) {
```

Comment:
Considered that but what arg key would we use? There's some risk that a user would stumble upon it and go down an unintended code path.

I suppose we could do something unlikely like `__amp_arg_expression`. This new param is pretty ugg-o.